pkg/catalogv2/oci/client.go

```diff
@@ -0,0 +1,251 @@
+package oci
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"helm.sh/helm/v3/pkg/chart/loader"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/helm/pkg/provenance"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+	helmregistry "helm.sh/helm/v3/pkg/registry"
+	"helm.sh/helm/v3/pkg/repo"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/memory"
+	"oras.land/oras-go/v2/registry"
+	orasRegistryRemote "oras.land/oras-go/v2/registry/remote"
+	orasRegistryAuth "oras.land/oras-go/v2/registry/remote/auth"
+)
+
+type OCIClient struct {
+	URL        string
+	repository string
+	registry   string
+	tag        string
+}
+
+// NewOCIClient returns a new OCIClient along
+// parsing the URL provided.
+func NewOCIClient(url string) (*OCIClient, error) {
+	ociClient := &OCIClient{
+		URL: url,
+	}
+
+	err := ociClient.parseURL()
+	if err != nil {
+		return nil, err
+	}
+
+	return ociClient, nil
+}
+
+func (o *OCIClient) parseURL() error {
+	url := strings.TrimSuffix(o.URL, "/")
+	url = strings.TrimPrefix(url, "oci://")
+
+	if strings.Contains(url, "/") {
+		var ref registry.Reference
+		ref, err := registry.ParseReference(url)
+		if err != nil {
+			return fmt.Errorf("failed to parse OCI URL '%s' value: %w", o.URL, err)
+		}
+		o.registry = ref.Registry
+		o.repository = ref.Repository
+		o.tag = ref.Reference
+	} else {
+		o.registry = url
+	}
+
+	return nil
+}
+
+func (o *OCIClient) fetchChart(orasRepository *orasRegistryRemote.Repository) (string, bool, error) {
```

Comment:
Please add a doc here describing the function

Reply:
Done.